js/modules/k6/html/html.go

```diff
@@ -71,3 +99,438 @@ func (s Selection) Attr(name string, def ...goja.Value) goja.Value {
+}
+
+func (s Selection) Parent(def ...string) Selection {
+	return s.adjacent(s.sel.Parent, s.sel.ParentFiltered, def...)
+}
+
+func (s Selection) Parents(def ...string) Selection {
+	return s.adjacent(s.sel.Parents, s.sel.ParentsFiltered, def...)
+}
+
+func (s Selection) Siblings(def ...string) Selection {
+	return s.adjacent(s.sel.Siblings, s.sel.SiblingsFiltered, def...)
+}
+
+func (s Selection) adjacentUntil(until func(string) *goquery.Selection,
+	untilSelection func(*goquery.Selection) *goquery.Selection,
+	filteredUntil func(string, string) *goquery.Selection,
+	filteredUntilSelection func(string, *goquery.Selection) *goquery.Selection,
+	def ...goja.Value) Selection {
+	// empty selector to nextuntil and prevuntil acts like revAll and nextAll
+	// relies on goquery.compileMatcher returning a matcher which fails all matches when the selector being compiled is invalid
+	if len(def) == 0 {
+		return Selection{s.rt, until("")}
+	}
+
+	selector := def[0].Export()
+	if len(def) == 1 {
+		switch selector.(type) {
+		case string:
+			return Selection{s.rt, until(selector.(string))}
+
+		case Selection:
+			return Selection{s.rt, untilSelection(selector.(Selection).sel)}
+
+		case nil:
+			return Selection{s.rt, until("")}
+
+		default:
+			panic(s.rt.NewGoError(errors.New("Invalid argument. The selector must be a string or query object")))
+			return Selection{}
+		}
+	} else {
+		filter := def[1].String()
+		switch selector.(type) {
+		case string:
+			return Selection{s.rt, filteredUntil(filter, selector.(string))}
+
+		case Selection:
+			return Selection{s.rt, filteredUntilSelection(filter, selector.(Selection).sel)}
+
+		case nil:
+			return Selection{s.rt, filteredUntil(filter, "")}
+
+		default:
+			panic(s.rt.NewGoError(errors.New("Invalid argument. The selector must be a string or query object")))
+			return Selection{}
+		}
+	}
+}
+
+// prevUntil, nextUntil and parentsUntil support two args
+// 1st arg is either a selector string, goquery selection object, or nil
+// 2nd arg is filter selector string or nil/undefined
+func (s Selection) PrevUntil(def ...goja.Value) Selection {
+	return s.adjacentUntil(
+		s.sel.PrevUntil,
+		s.sel.PrevUntilSelection,
+		s.sel.PrevFilteredUntil,
+		s.sel.PrevFilteredUntilSelection,
+		def...,
+	)
+}
+
+func (s Selection) NextUntil(def ...goja.Value) Selection {
+	return s.adjacentUntil(
+		s.sel.NextUntil,
+		s.sel.NextUntilSelection,
+		s.sel.NextFilteredUntil,
+		s.sel.NextFilteredUntilSelection,
+		def...,
+	)
+}
+
+func (s Selection) ParentsUntil(def ...goja.Value) Selection {
+	return s.adjacentUntil(
+		s.sel.ParentsUntil,
+		s.sel.ParentsUntilSelection,
+		s.sel.ParentsFilteredUntil,
+		s.sel.ParentsFilteredUntilSelection,
+		def...,
+	)
+}
+
+func (s Selection) Slice(start int, def ...int) Selection {
+	if len(def) > 0 {
+		return Selection{s.rt, s.sel.Slice(start, def[0])}
+	} else {
+		return Selection{s.rt, s.sel.Slice(start, s.sel.Length())}
```

Comment:
Just have this the default. It kinda makes sense to have it as an if/else, so I'm torn between the two, but my gut tells me it would be better as a default flow. It's that or a `switch len(def) {}` with only 0/default cases and that feels like it would look downright odd...